process/phpcs/phpcompatibility.go

```diff
@@ -0,0 +1,98 @@
+package phpcs
+
+import (
+	"io/ioutil"
+	"github.com/wptide/pkg/phpcompat"
+	"github.com/wptide/pkg/tide"
+)
+
+var (
+	writeFile = ioutil.WriteFile
+)
+
+type PhpCompatDetails struct {
+	Totals   map[string]int                       `json:"totals"`
+	ErrorMap map[string][]string                  `json:"error_map"`
+	Errors   map[string]PhpCompatDetailsViolation `json:"errors"`
+}
+
+type PhpCompatDetailsViolation struct {
+	Message  string                    `json:"message"`
+	Source   string                    `json:"source"`
+	Type     string                    `json:"type"`
+	Severity int                       `json:"severity"`
+	Versions []string                  `json:"versions"`
+	Files    map[string][]FilePosition `json:"files"`
+}
+
+type FilePosition struct {
+	Line   int `json:"line"`
+	Column int `json:"column"`
+}
+
+// GetPhpcsCompatibility runs the PHPCompatiblity post processing and determines the compatible versions.
+//
+// A detailed report is also sent to a storage provider which contains a structure ordered by each violating sniff which gives:
+// - the PHP versions effected by the violation
+// - the impacted files and relevant phpcs messages
+//
+// Process is required to implement audit.PostProcessor.
+func GetPhpcsCompatibility(fullResults tide.PhpcsResults) ([]string, interface{}) {
+
+	brokenVersions := []string{}
+
+	// Dynamically creating our struct for JSON output.
+	details := &PhpCompatDetails{
+		Totals: map[string]int{
+			"errors":   fullResults.Totals.Errors,
+			"warnings": fullResults.Totals.Warnings,
+		},
+		ErrorMap: make(map[string][]string),
+		Errors:   make(map[string]PhpCompatDetailsViolation),
```

Comment:
We should add warnings in since the UI for .org has them referenced in the mockups.

Reply:
Can you create an issue?  I'd rather it in a separate PR so that I can focus on getting the services going again.